### PR TITLE
Refs #54570 gem ransack search books

### PR DESCRIPTION
**Pull request by ggbond2605**

### CODE REVIEW

File: app/assets/javascripts/book.coffee [hunk 1]
```diff
@@ -0,0 +1,10 @@
+jQuery ->
+  $('form').on 'click', '.remove_fields', (event) ->
```
Comment (NgocVu1994) [inline app/assets/javascripts/book.coffee]: " "

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -3,4 +3,13 @@ def full_title page_title = ""
     base_title = t :system_name
     page_title.empty? ? base_title : "#{page_title} | #{base_title}"
   end
+
+  def link_to_add_fields(name, f, type)
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: remove ()

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -3,4 +3,13 @@ def full_title page_title = ""
...
+    new_object = f.object.send "build_#{type}"
+    id = "new_#{type}"
+    fields = f.send("#{type}_fields", new_object, child_index: id) do |builder|
+      render(type.to_s + "_fields", f: builder)
+    end
+    link_to(name, "#", class: "add_fields", data: {id: id, fields: fields.gsub("\n", "")})
```
Comment (hellovietnam93) [inline app/helpers/application_helper.rb]: remove ()

File: app/models/book.rb [hunk 4]
```diff
@@ -11,4 +11,10 @@ class Book < ActiveRecord::Base
   validates :number_of_pages, numericality: {only_integer: true}
 
   enum status: [:reading, :read]
+
+  class << self
+    def self.ransackable_attributes(auth_object = nil)
```
Comment (hellovietnam93) [inline app/models/book.rb]: self trong self?

File: app/views/books/_condition_fields.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  <%= f.attribute_fields do |a| %>
+    <%= a.attribute_select %>
+  <% end %>
+  <%= f.predicate_select :only => [:cont, :eq], compounds: false %>
+  <%= f.value_fields do |v| %>
+    <%= v.text_field :value %>
+  <% end %>
+  <%= link_to t(:remove), "#", class: "remove_fields" %>
+</div>
+
```
Comment (hellovietnam93) [inline app/views/books/_condition_fields.html.erb]: remove ()

File: app/views/books/_condition_fields.html.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<div class="field">
+  <%= f.attribute_fields do |a| %>
+    <%= a.attribute_select %>
+  <% end %>
+  <%= f.predicate_select :only => [:cont, :eq], compounds: false %>
```
Comment (hellovietnam93) [inline app/views/books/_condition_fields.html.erb]: convention

File: app/models/book.rb [hunk 7]
```diff
@@ -11,4 +11,10 @@ class Book < ActiveRecord::Base
   validates :number_of_pages, numericality: {only_integer: true}
 
   enum status: [:reading, :read]
+
+  class << self
+    def ransackable_attributes(auth_object = nil)
```
Comment (hellovietnam93) [inline app/models/book.rb]: remove ()

**Comment by hellovietnam93** [comment]
OK

File: app/controllers/books_controller.rb [hunk 8]
```diff
@@ -2,7 +2,8 @@ class BooksController < ApplicationController
   load_and_authorize_resource
 
   def index
-    @books = Book.paginate page: params[:page], per_page: Settings.books.page
+    @search = Book.search params[:q]
+    @books = @search.result(distinct: true).paginate page: params[:page], per_page: Settings.books.page
```
Comment (dieunb) [inline app/controllers/books_controller.rb]: dài quá 80 ký tự

**Pull request merged**